lib/XMLHttpRequest.js

```diff
@@ -377,6 +435,9 @@ function XMLHttpRequest(opts) {
         host = url.hostname;
         break;
 
+      case 'data:':
+        datauri = true;
+      
```

Comment:
I don't think multiple protocols can be valid at the same time, so a `break` is missing here

Reply:
for data URI i parse the URL inside `local` check like this:
```js
if (local) {
	if (datauri) {
		// parse Data URI
		return;
	}
	// parse file protocol
	return;
}
```

so a non-break here is intentional, as the next step it marks the `local` flag, and allows the data URI to be parsed like in the logic above